kerfuffle/archiveentry.h

```diff
@@ -42,24 +43,26 @@ class Archive::Entry : public QObject
     Q_PROPERTY(QString method MEMBER m_method)
     Q_PROPERTY(QString version MEMBER m_version)
     Q_PROPERTY(QDateTime timestamp MEMBER m_timestamp)
-    Q_PROPERTY(bool isDirectory MEMBER m_isDirectory)
+    Q_PROPERTY(bool isDirectory MEMBER m_isDirectory WRITE setIsDirectory)
     Q_PROPERTY(QString comment MEMBER m_comment)
     Q_PROPERTY(bool isPasswordProtected MEMBER m_isPasswordProtected)
 
 public:
 
-    Entry(Entry *parent);
+    Entry(Entry *parent, QString fullPath = QString(), QString rootNode = QString());
     ~Entry();
 
     QList<Entry*> entries();
+    const QList<Entry*> entries() const;
```

Comment:
Call this method `children()` (and also `m_entries` => `m_children`).

Also, what about storing `const Entry*` in the list? I don't think we want to allow editing children entries by using this method...